substrate/frame/treasury/src/lib.rs

```diff
@@ -361,14 +358,10 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config<I>, I: 'static = ()> {
-		/// New proposal.
-		Proposed { proposal_index: ProposalIndex },
```

Comment:
we have implicit codec indexes here, since we need to remove some variants now, we should set them explicitly so that the remaining variants keep the same indexes as before. this should apply for `Event` and `Error`. something like codec(index = 1) for `Spending`

Reply:
while it is good to keep the index, it is not critical. events should not be stored in any onchain storages and for offchain decode, we always use metadata to identify the variant

Reply:
@muharem I pushed this new commit: https://github.com/paritytech/polkadot-sdk/pull/3820/commits/2e8b334173ace4f8c666ffae1a53583efc4b3a91

This is not fully implemented as I need to calculate the index of 
```diff
+ #[codec(index = 4)]
Burnt { burnt_funds: BalanceOf<T, I> },
```
If we follow the feedback from @xlc, I can remove the added code.

Reply:
@muharem I removed the code

Reply:
@chungquantin please check this comment https://github.com/paritytech/polkadot-sdk/pull/3820#issuecomment-2020821759